interfaces/apparmor/backend.go

```diff
@@ -309,7 +309,7 @@ func (b *Backend) Setup(snapInfo *snap.Info, opts interfaces.ConfinementOptions,
 	// TODO: we need to deal with the "snapd" snap here soon
 	if snapName == "core" && release.OnClassic && release.AppArmorLevel() != release.NoAppArmor {
 		if err := setupSnapConfineReexec(snapInfo); err != nil {
-			logger.Noticef("cannot create host snap-confine apparmor configuration: %s", err)
+			return fmt.Errorf("cannot create host snap-confine apparmor configuration: %s", err)
```

Comment:
Just to be on the safe side - I think we should add a spread test that does something like:
```
systemctl stop snapd.socket snapd
mount -o bind /bin/false /sbin/apparmor_parser
systemctl start snapd snapd.socket
journalctl -u snapd | MATCH "cannot regenerate apparmor profile"
snap list | MATCH core
```
to ensure failures in the apparmor (re)generation do not prevent snapd from starting.

One aspect of this change makes me slightly nervous - it may prevent core refreshes. This can be a good thing, i.e. if we would write something that is unusable but we need to be mindful about it. Its one of the few thngs that can prevent a core refresh.

Reply:
+1 but I will do this in https://github.com/snapcore/snapd/pull/6233 where this feature is "landing", then I will back port here to ensure it all works in the 2.36 world.

Reply:
6233 doesn't have this change, or was it landed separately and to visible there?

Reply:
I can back port all the tests as well. Let me do that now.

Reply:
I've back ported all the things from https://github.com/snapcore/snapd/pull/6233